### PR TITLE
demo: update 4.x dropdown deprecated menu

### PR DESCRIPTION
**Pull request by linxianxi**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [x] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

deprecated demo 应该使用 deprecated 语法

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |      -     |
| 🇨🇳 Chinese |    -       |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at d9b9f73</samp>

Updated `Menu` component to use new API and fixed a typo in `components/dropdown/demo/deprecated.md`. The change removed the `items` prop and improved the menu rendering.

### <samp>🤖 Generated by Copilot at d9b9f73</samp>

*  Removed the deprecated `items` prop from the `Menu` component and replaced it with the `children` prop, which accepts an array of `Menu.Item` or `Menu.SubMenu` elements. This allows more flexibility and consistency in rendering menu items and submenus.

### CODE REVIEW

File: components/dropdown/demo/deprecated.md [hunk 1]
```diff
@@ -20,42 +20,24 @@ import { Dropdown, Menu, Space } from 'antd';
...
-      {
-        key: '4',
-        danger: true,
-        label: 'a danger item',
-      },
-    ]}
-  />
+  <Menu>
+    <Menu.Item>
+      <a target="_blank" rel="noopener noreferrer" href="http://www.antgroup.com/">
```
Comment (MadCcc) [inline components/dropdown/demo/deprecated.md:25]:
```suggestion
      <a target="_blank" rel="noopener noreferrer" href="https://www.antgroup.com/">
```

File: components/dropdown/demo/deprecated.md [hunk 2]
```diff
@@ -20,42 +20,24 @@ import { Dropdown, Menu, Space } from 'antd';
...
+        1st menu item
+      </a>
+    </Menu.Item>
+    <Menu.Item disabled icon={<SmileOutlined />}>
+      <a target="_blank" rel="noopener noreferrer" href="http://www.aliyun.com/">
```
Comment (MadCcc) [inline components/dropdown/demo/deprecated.md:30]:
```suggestion
      <a target="_blank" rel="noopener noreferrer" href="https://www.aliyun.com/">
```

File: components/dropdown/demo/deprecated.md [hunk 3]
```diff
@@ -20,42 +20,24 @@ import { Dropdown, Menu, Space } from 'antd';
...
+        2rd menu item (disabled)
+      </a>
+    </Menu.Item>
+    <Menu.Item disabled>
+      <a target="_blank" rel="noopener noreferrer" href="http://www.luohanacademy.com/">
```
Comment (MadCcc) [inline components/dropdown/demo/deprecated.md:35]:
```suggestion
      <a target="_blank" rel="noopener noreferrer" href="https://www.luohanacademy.com/">
```

**Comment by linxianxi** [comment]
ci 挂了，应该跟我这个没关系

File: components/dropdown/__tests__/__snapshots__/demo-extend.test.ts.snap [hunk 4]
```diff
@@ -3468,11 +3468,11 @@ Array [
             class="ant-dropdown-menu-title-content"
           >
             <a
-              href="https://www.aliyun.com"
+              href="https://www.aliyun.com/"
               rel="noopener noreferrer"
               target="_blank"
             >
-              2nd menu item (disabled)
+              2rd menu item (disabled)
```
Comment (afc163) [inline components/dropdown/__tests__/__snapshots__/demo-extend.test.ts.snap:3475]: 2nd is correct usage

File: components/dropdown/__tests__/__snapshots__/demo-extend.test.ts.snap [hunk 5]
```diff
@@ -3410,7 +3410,7 @@ Array [
             class="ant-dropdown-menu-title-content"
           >
             <a
-              href="https://www.antgroup.com"
+              href="https://www.antgroup.com/"
```
Comment (afc163) [inline components/dropdown/__tests__/__snapshots__/demo-extend.test.ts.snap:3413]: 不需要动这个。

**Review by afc163 — APPROVED** [review]

**Review by afc163 — APPROVED** [review]

**Pull request merged**